initial commit with HA details (#2423)
astronomer/docs: astro/configure-deployment-resources.md

## Enable high availability

By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of its scheduler across different nodes. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Note that PGBouncer Pods are guaranteed to run across multiple availability zones, but scheduler Pods are not. 

Because this setting results in more resource usage, it can increase the cost of your Deployment. See [Pricing](https://astronomer.io/pricing).

1. In the Cloud UI, select a Workspace, click **Deployments**, and then select a Deployment.
2. Click the **Details** tab.
3. Click **Edit Details**.
4. In **High Availability**, click the toggle to **On**.

:::info Alternative Astro Hybrid Setup

> On Astro Hybrid, PgBouncer is highly available by default for your Deployment whereas to enable high availability of schedulers you need to configure your scheduler count to more than 1. 

Reviewer: ```suggestion
On Astro Hybrid, PgBouncer is highly available by default for all Deployments. Schedulers are highly available if a Deployment uses two or more schedulers. 
```

> Every Deployment has 2 PgBouncer Pods assigned to two different nodes to prevent zombie tasks. If you configure your Deployment with 2 schedulers, each scheduler Pod is assigned to a separate node to ensure availability. To limit cost, a Deployment that uses 3 or 4 schedulers can assign all scheduler Pods across two nodes.

Reviewer: ```suggestion
Every Deployment has two PgBouncer Pods assigned to two different nodes to prevent zombie tasks. If you configure your Deployment with two schedulers, each scheduler Pod is assigned to a separate node to ensure availability. To limit cost, a Deployment that uses three or four schedulers can assign all scheduler Pods across two nodes.
```

## Enable high availability

:::info Hybrid high availability

**This setting is configurable only on Astro Hosted Deployments. **

> On Astro Hybrid, by default, PgBouncer is highly available. Every Deployment has 2 PgBouncer pods each assigned to a different node. This prevents against zombie tasks. PgBouncer is not configurable and is managed by Astronomer. 

Author: Not sure why pods should start with capital P ; it is not an Astro term but a generic K8s term? I changed the edits you made because that changed the meaning. if we use "each node" that could imply each node will have a pgbouncer pod.

Each Deployment has only one scheduler by default. But users can configure their Deployment to use up to 4 schedulers. For users who set Scheduler Count in the Cloud UI to 2, this means that those 2 scheduler Pods cannot be assigned to the same node and instead require a minimum of 2 nodes total. To avoid significant increases in cost, 3 or 4 schedulers can share the same 2 nodes and will not necessarily result in a higher node count.
:::

By default, all of your Airflow components have one instance running which are spread across multiple nodes. When you enable high availability for a Deployment, it runs two instances of [PgBouncer](https://www.pgbouncer.org/) and scheduler on different nodes for resiliency. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Because this setting results in more resource usage, it can increase the cost of your Deployment. See [Pricing](https://astronomer.io/pricing).

1. In the Cloud UI, select a Workspace, click **Deployments**, and then select a Deployment.
2. Click the **Details** tab.
3. Click **Edit Details**.
4. In **High Availability**, click the toggle to **On**.

> *Multiple instances of [PgBouncer](https://www.pgbouncer.org/) run across availability zones; for scheduler it is not guaranteed but is possible because the cluster is across two availability zones. Support for scheduler is coming soon*

Author: I thought we were avoiding, but then read point 7 in [this](https://docs.astronomer.io/astro/cloud-ide/deploy-project#export-your-pipelines-to-a-local-astro-project
)

> *Multiple instances of [PgBouncer](https://www.pgbouncer.org/) run across availability zones; for scheduler it is not guaranteed but is possible because the cluster is across two availability zones. Support for scheduler is coming soon*

Reviewer: That should definitely be deleted 😓 not sure how I let that through review!

> ## Enable high availability

Reviewer 2: From standup:

- Let's consider making this part of the doc assume Hosted only
- Add info on Hybrid HA in the cluster settings resource reference
- Say something like, "If you're an Astro Hybrid customer, HA is... see here for more info"

## Enable high availability

> :::info High availability on Astro Hybrid

Reviewer: We should move this tip until after the numbered steps for Hosted

High availability is configurable only on Astro Hosted Deployments.

On Astro Hybrid, PgBouncer is highly available by default. Every Deployment has 2 PgBouncer Pods assigned to two different nodes to prevent zombie tasks.

> By default, each deployment uses 1 scheduler. If you configure your Deployment to use 2 schedulers, each scheduler Pod is assigned to a separate node to ensure availability. To limit in cost, a Deployment that uses 3 or 4 schedulers can assign all scheduler Pods across two nodes.

Reviewer: ```suggestion
By default, each Deployment uses 1 scheduler. If you configure your Deployment with 2 schedulers, each scheduler Pod is assigned to a separate node to ensure availability. To limit cost, a Deployment that uses 3 or 4 schedulers can assign all scheduler Pods across two nodes.
```

:::

> By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of scheduler across different nodes. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Note that PGBouncer Pods are guaranteed to run across multiple availability zones, but scheduler Pods are not. 

Reviewer: ```suggestion
By default, the Pods running your Deployment's Airflow components are distributed across multiple nodes. When you enable high availability, your Deployment runs two instances of [PgBouncer](https://www.pgbouncer.org/) and two instances of its scheduler across different nodes. This ensures that your DAGs can continue to run if there's an issue with one of your Airflow components in a specific node. Note that PGBouncer Pods are guaranteed to run across multiple availability zones, but scheduler Pods are not. 
```
